pandas/tseries/tests/test_tslib.py

```diff
@@ -180,6 +180,63 @@ def test_constructor_invalid(self):
         with tm.assertRaisesRegexp(ValueError, 'Cannot convert Period'):
             Timestamp(Period('1000-01-01'))
 
+    def test_constructor_positional(self):
+        # GH 10758
+        with tm.assertRaises(TypeError):
+            Timestamp(2000, 1)
```

Comment:
For the cases where not enough parameters are passed to `Timestamp`, the error will be `an integer is required (got NoneType)` instead of `Required argument not found`. Matching the `datetime` exception exactly requires 3 additional checks and calls:

```
if year is None:
    return datetime.datetime()
if month is None:
    return datetime.datetime(year)
if day is None:
    return datetime.datetime(year, month)
```

which seems ugly. Do we really want that?